entry-generation/godot-compiler-plugin/src/main/kotlin/godot/compilerplugin/CommonComponentRegistrar.kt

```diff
@@ -0,0 +1,59 @@
+package godot.compilerplugin
+
+import com.intellij.mock.MockProject
+import de.jensklingenberg.mpapt.common.MpAptProject
+import godot.annotation.processor.GodotAnnotationProcessor
+import org.jetbrains.kotlin.codegen.extensions.ClassBuilderInterceptorExtension
+import org.jetbrains.kotlin.compiler.plugin.*
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.extensions.StorageComponentContainerContributor
+import org.jetbrains.kotlin.js.translate.extensions.JsSyntheticTranslateExtension
+
+class CommonComponentRegistrar : ComponentRegistrar {
+    override fun registerProjectComponents(
+        project: MockProject,
+        configuration: CompilerConfiguration
+    ) {
+        val processor = GodotAnnotationProcessor(
+            configuration.get(CompilerPluginConst.CommandlineArguments.ENTRY_DIR_PATH)!!,
```

Comment:
use `checkNotNull(...) { "error message here" }`, `!!` just throws an NPE which is not really helpful for debugging.

Reply:
True! Good idea